test/parallel/test-zlib-dictionary.js

```diff
@@ -58,23 +60,84 @@ function deflateResetDictionaryTest() {
   let output = '';
   const deflate = zlib.createDeflate({ dictionary: spdyDict });
   const inflate = zlib.createInflate({ dictionary: spdyDict });
+  inflate.setEncoding('utf-8');
 
-  deflate.on('data', function(chunk) {
+  // Wrapper+data for each round of input => 4 calls
+  deflate.on('data', common.mustCall(function(chunk) {
     if (doneReset)
       inflate.write(chunk);
-  });
+  }, 4));
 
-  inflate.on('data', function(chunk) {
+  inflate.on('data', common.mustCall(function(chunk) {
     output += chunk;
-  });
+  }));
 
-  deflate.on('end', function() {
+  deflate.on('end', common.mustCall(function() {
     inflate.end();
-  });
+  }));
+
+  inflate.on('end', common.mustCall(function() {
+    assert.strictEqual(input, output);
+  }));
 
-  inflate.on('end', function() {
-    assert.equal(input, output);
+  deflate.write(input);
+  deflate.flush(function() {
+    deflate.reset();
+    doneReset = true;
+    deflate.write(input);
+    deflate.end();
   });
+}
+
+function rawDictionaryTest() {
+  let output = '';
+  const deflate = zlib.createDeflateRaw({ dictionary: spdyDict });
+  const inflate = zlib.createInflateRaw({ dictionary: spdyDict });
+  inflate.setEncoding('utf-8');
+
+  deflate.on('data', common.mustCall(function(chunk) {
+    inflate.write(chunk);
+  }));
+
+  inflate.on('data', common.mustCall(function(chunk) {
+    output += chunk;
+  }));
+
+  deflate.on('end', common.mustCall(function() {
+    inflate.end();
+  }));
+
+  inflate.on('end', common.mustCall(function() {
+    assert.strictEqual(input, output);
+  }));
+
+  deflate.write(input);
+  deflate.end();
+}
+
+function deflateRawResetDictionaryTest() {
+  let doneReset = false;
+  let output = '';
+  const deflate = zlib.createDeflateRaw({ dictionary: spdyDict });
+  const inflate = zlib.createInflateRaw({ dictionary: spdyDict });
+  inflate.setEncoding('utf-8');
+
+  deflate.on('data', common.mustCall(function(chunk) {
+    if (doneReset)
+      inflate.write(chunk);
+  }, 2));
```

Comment:
I’d say `common.mustCall()` for `data` callbacks is optional, I don’t think we’re currently guaranteeing anything about the exact chunking or want to do that (I think this is okay if the CI is green with it included, so it’s your call).